### PR TITLE
Use FPM instead of NodePackage to build packages

### PR DESCRIPTION
**Pull request by dergraf**
Install FPM: `gem install fpm`
Run `make deb` or `make rpm`

### CODE REVIEW

**Comment by codeadict** [comment]
Also probably worth looking at https://github.com/rabbitmq/rabbitmq-server-release

**Comment by dergraf** [comment]
- Checked that the RPM can be built and installed on Centos7 and OpenSuse
- Checked that the RPM can be build and installed on current Debian Testing

File: Makefile [hunk 1]
```diff
@@ -1,231 +1,148 @@
...
+		_build/default/rel/vernemq/log/=/var/log/vernemq/
+
+pkg_clean:
+	rm -rf _build/default/rel
 
 ##
 ## Developer targets
 ##
 ##  devN - Make a dev build for node N
 dev% :
```
Comment (larshesel) [inline Makefile]: The dev targets doesn't seem to work.

I tried doing `make dev1 dev2 dev3` and then when starting I get:

```
$ _build/dev1/rel/vernemq/bin/vernemq start
!!!!
!!!! WARNING: ulimit -n is 1024; 65536 is the recommended minimum.
!!!!
vmq_cluster_node_sup
vernemq failed to start within 15 seconds,
see the output of 'vernemq console' for more information.
If you want to wait longer, set the environment variable
WAIT_FOR_ERLANG to the number of seconds to wait.
```

And it doesn't look like vernemq was started.

File: Makefile [hunk 2]
```diff
@@ -1,231 +1,148 @@
...
+
+package: $(PKGTARGET)
+
+pkg_rel: pkg_clean
+	$(MAKE) rel
+
+pkg_strip:
+	$(shell find _build/default/rel/vernemq -type f -executable | xargs strip --remove-section=.comment --remove-section=.note --strip-unneeded)
+
+deb: OVERLAY_VARS=vars/deb_vars.config
```
Comment (larshesel) [inline Makefile]: make deb
rm -rf _build/default/rel
make rel
make[1]: Entering directory '/home/lhc/dev/octavolabs/vernemq'
cat vars.config > vars.generated
cat vars/deb_vars.config >> vars.generated
/home/lhc/dev/octavolabs/vernemq/rebar3  release
===> Verifying dependencies...
===> Compiling vmq_ql
===> Running neotoma...
===> Compiling vmq_plugin
===> Compiling vmq_commons
===> Compiling vmq_server
===> Compiling vmq_webhooks
===> Compiling vmq_swc
===> Compiling vmq_acl
===> Compiling vmq_plumtree
===> Compiling vmq_passwd
make[2]: Entering directory '/home/lhc/dev/octavolabs/vernemq/apps/vmq_passwd/c_src'
cc   vmq_passwd.c -lcrypto -o ../priv/vmq_passwd
make[2]: Leaving directory '/home/lhc/dev/octavolabs/vernemq/apps/vmq_passwd/c_src'
===> Compiling vmq_mqtt5_demo_plugin
===> Compiling vmq_bridge
===> Compiling vmq_diversity
===> Running cuttlefish schema generator
===> Starting relx build process ...
===> Resolving OTP Applications from directories:
          /home/lhc/dev/octavolabs/vernemq/_build/default/lib
          /home/lhc/dev/octavolabs/vernemq/apps
          /home/opt/erlang/erlang21.3.1/lib
===> Resolved vernemq-1.7.0+build.34.ref4174d018
===> Including Erts from /home/opt/erlang/erlang21.3.1
===> release successfully created!
make[1]: Leaving directory '/home/lhc/dev/octavolabs/vernemq'
strip:_build/default/rel/vernemq/bin/cuttlefish: file format not recognized
strip:_build/default/rel/vernemq/bin/vmq-admin: file format not recognized
strip:_build/default/rel/vernemq/bin/vernemq: file format not recognized
strip:_build/default/rel/vernemq/lib/inets-7.0.6/priv/bin/runcgi.sh: file format not recognized
strip:_build/default/rel/vernemq/lib/env.sh: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/cuttlefish: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/start_erl.src: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/nodetool: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/erl.src: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/start: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/erl: file format not recognized
strip:_build/default/rel/vernemq/erts-10.3/bin/start.src: file format not recognized
dpkg-shlibdeps: error: cannot read debian/control: No such file or directory
xargs: dpkg-shlibdeps: exited with status 255; aborting
# required for running dpkg-shlibdeps
mkdir -p debian
touch debian/control
fpm -s dir -t deb -v "1.7.0-34-g4174d018" \
	--force \
	--name vernemq \
	--epoch 1553699570 \
	--license "Apache 2.0" \
	--url "https://vernemq.com" \
	--vendor "Octavo Labs AG" \
	--maintainer "<info@vernemq.com>" \
	--description "VerneMQ is a MQTT message broker" \
	--depends logrotate \
	--depends sudo \
	--depends adduser \
	--depends "" \
	--deb-user vernemq \
	--deb-group vernemq \
	--deb-changelog ./changelog.md \
	--deb-no-default-config-files \
	--deb-systemd files/vernemq.service \
	--after-install files/deb-vernemq.postinst \
	--config-files /etc/vernemq/vernemq.conf \
	_build/default/rel/vernemq/bin/vernemq=/usr/bin/vernemq \
	_build/default/rel/vernemq/bin/vmq-admin=/usr/bin/vmq-admin \
	_build/default/rel/vernemq/data=/var/lib/vernemq/ \
	_build/default/rel/vernemq/etc/=/etc/vernemq/ \
	_build/default/rel/vernemq/bin/=/usr/lib/vernemq/bin/ \
	_build/default/rel/vernemq/lib=/usr/lib/vernemq/ \
	_build/default/rel/vernemq/releases=/usr/lib/vernemq/ \
	_build/default/rel/vernemq/erts-"10.3"=/usr/lib/vernemq/ \
	_build/default/rel/vernemq/share/=/usr/share/vernemq/ \
	_build/default/rel/vernemq/log/=/var/log/vernemq/
Adding action files
epoch in Version is set {:epoch=>"1553699570", :level=>:warn}
Created package {:path=>"vernemq_1.7.0-34-g4174d018_amd64.deb"}
21.3.1 lhc@algol ~/dev/octavolabs/vernemq (use-fpm-instead-of-nodepackage) $ sudo dpkg -i ./vernemq_1.7.0-34-g4174d018_amd64.deb 
[sudo] password for lhc: 
dpkg: error processing archive ./vernemq_1.7.0-34-g4174d018_amd64.deb (--install):
 parsing file '/var/lib/dpkg/tmp.ci/control' near line 8 package 'vernemq':
 'Depends' field, missing package name, or garbage where package name expected
Errors were encountered while processing:
 ./vernemq_1.7.0-34-g4174d018_amd64.deb

**Pull request merged**